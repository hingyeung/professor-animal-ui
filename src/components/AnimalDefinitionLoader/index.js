import AnimalDefinitionLoader from './AnimalDefinitionLoader.jsx';

export default AnimalDefinitionLoader;