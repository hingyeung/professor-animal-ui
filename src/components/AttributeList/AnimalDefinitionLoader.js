import React from 'react';

const AnimalDefinitionLoader = (props) => {
  function onChange(event) {
    const fileToOpen = event.target.files[0];

    if (fileToOpen && fileToOpen.type === 'application/json') {
      let fileReader = new FileReader();

      fileReader.onload = function(e) {
        let jsonStr = fileReader.result;
        props.onFileLoaded(jsonStr);
      };

      fileReader.readAsText(fileToOpen);
    }
  }

  return (
    <div>
      <form>
        <input type="file" className="form-control" onChange={onChange}/>
      </form>
    </div>
  );
};

export default AnimalDefinitionLoader;