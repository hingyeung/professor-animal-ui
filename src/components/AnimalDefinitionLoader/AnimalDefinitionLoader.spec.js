import React from 'react';
import AnimalDefinitionLoader from "./AnimalDefinitionLoader";
import {shallow} from 'enzyme';

describe('AnimalDefinitionLoader', function () {
  const mockOnFileLoaded = jest.fn();

  it('should render properly', function () {
    const wrapper = shallow(<AnimalDefinitionLoader onFileLoaded={mockOnFileLoaded}/>);
    expect(wrapper).toMatchSnapshot();
  });

  it('should call onFileLoaded in props on change', function () {
    const dummyFileReader = {
      readAsText: function() {this.onload()},
      onload: function() {},
      result: 'file content'
    };
    const mockFile = {name: 'somefile', type: 'application/json'};
    const originalFileReader = window.FileReader;
    window.FileReader = jest.fn(() => dummyFileReader);

    const wrapper = shallow(<AnimalDefinitionLoader onFileLoaded={mockOnFileLoaded}/>);
    wrapper.find('input').simulate('change', {target: {files: [mockFile]}});

    expect(mockOnFileLoaded).toHaveBeenCalledTimes(1);
    expect(mockOnFileLoaded).toHaveBeenCalledWith('file content');

    // restore FileReader
    window.FileReader = originalFileReader;
  })
});