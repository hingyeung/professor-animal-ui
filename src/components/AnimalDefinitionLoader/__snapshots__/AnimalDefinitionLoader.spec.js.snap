// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnimalDefinitionLoader should render properly 1`] = `
<div>
  <form>
    <input
      className="form-control"
      onChange={[Function]}
      type="file"
    />
  </form>
</div>
`;
